App/BitBar/ExecutablePlugin.m

```diff
@@ -70,86 +70,97 @@ -(void)performRefreshNow:(NSMenuItem*)menuItem{
 }
 
 -(BOOL)refresh {
+  __unsafe_unretained ExecutablePlugin *weakSelf = self;
   [self.lineCycleTimer invalidate];
   self.lineCycleTimer = nil;
   [self.refreshTimer invalidate];
   self.refreshTimer = nil;
-    
+
   // execute command
   dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT,0),  ^{
-    [self refreshContentByExecutingCommand];
+    [weakSelf refreshContentByExecutingCommand];
     dispatch_sync(dispatch_get_main_queue(), ^{
-      
-      self.lastUpdated = NSDate.new;
-      
-      [self rebuildMenuForStatusItem:self.statusItem];
-      
-      // reset the current line
-      self.currentLine = -1;
-      
-      // update the status item
-      [self cycleLines];
-      
-      // sort out multi-line cycler
-      if (self.isMultiline) {
+      if (weakSelf) {
+        __strong ExecutablePlugin* strongSelf = weakSelf;
```

Comment:
No idea to be honest.  Nothing executes after it so I see no reason to wait for the block to return.  Maybe @matryer can say for certain :)

However, if it was changed to `dispatch_async`, I believe you'd likely need a separate strong reference to weakSelf...  (ie back to the code I had above.)